NavBar Links shown appropriately based off logged in status and active based off page loaded.

leechristophermurray opened the pull request:
The list of features this branch introduces are as follows.
* based off the page loaded, the appropriate navbar link will be highlighted
    * _to facilitate this, the **docs is now loaded as an iframe**._
    * changed the height to 88% in the style.css to remove the double scrollbars on the documentation page.
* `Harvest API` link now returns to the homepage
* ~~`Dashboard`~~ `My Apps` link is now shown only when logged in. This was implemented by adding `ng-if="userLoggedIn"` to the link tags attributes. *The developer dashboard is now labeled as **"My Apps"** as requested on #89.*
*  `Dashboard` link is now a link to the admin dashboard and shown **only** when and admin user is logged in. This was implemented by adding `ng-if="current_user.us_user_role == 'admin'"` to the link tags attributes.

This: 
* fixes #96
* fixes #50 
* fixes #56 
* fixes #49 
* fixes #89

**THIS BRANCH FIXES ALL THE ISSUES RELATED TO THE NAVBAR**

leechristophermurray committed:
The navbar links will now only reveal 'Dashboard' for a logged in user

leechristophermurray committed:
Update db.js

reverted mongoose.connect() arguments.

leechristophermurray committed:
fixed the active link items in the navbar

leechristophermurray committed:
the active links for the nav bar now works for the documentation. This was done by making the documentation load as an iframe

leechristophermurray committed:
Merge branch 'navbar-links-visibility' of github.com:slashroots/harvest-v2 into navbar-links-visibility

leechristophermurray committed:
changed the height to 88% in the style.css to remove the double scrollbars on the docubemtation page.

matjames007 commented on views/docs.ejs:
```diff
@@ -30,12 +30,12 @@
 
```
@thinkking is this file being used? -if not remove it

leechristophermurray replied:
Yes, it is. The original `/docs` path to the documentation is still available but is now being used to load the docs as an iframe as described in the pull request (#133) description. This had to be done to facilitate the docs' tab being shown as active as angular will not detect a path outside the app.

matjames007 commented on public/javascript/controllers/user-ctrl.js:
```diff
@@ -202,6 +226,10 @@ angular.module('harvestv2')
                 });
             };
 
+            $scope.isActive = function (linkPath) {
```
@thinkking What does this do?  Comment the code.

matjames007 commented on public/javascript/controllers/user-ctrl.js:
```diff
@@ -86,6 +86,30 @@ angular.module('harvestv2')
+).controller("DocsCtrl", ['$scope', '$location', '$routeParams', 'UserFactory', 'RolesFactory',
+        'PlatformFactory', 'CurrentUserFactory',
+        function($scope, $location, $routeParams, UserFactory, RolesFactory, PlatformFactory, CurrentUserFactory) {
+
```
@thinkking writing jQuery within angular framework is bad.  Attempt to rewrite this controller using Angular functions.

matjames007 commented on models/db.js:
```diff
@@ -6,7 +6,7 @@
 var Schema = mongoose.Schema;
 
-mongoose.connect(process.env.MONGOLAB_URI);
+mongoose.connect(process.env.MONGOLAB_URI || "localhost:27017/mongodb");
```
@thinkking change this.

matjames007 reviewed (changes requested):
Comments made within the in-line commenting feature.  Please review.

leechristophermurray committed:
The navbar links have changed. 'Dashboard', 'My Apps', 'Documentation', and 'ABIS Home Page' are now the links. 'Dashboard' links to the admin dashboard. 'My Apps' links to what was previously the developer dashboard. The rest od the navbar links' functionality remain the same. It is noteworthy that the 'Dashboard' link will only show when a user with 'current_user.us_user_role' being 'admin' is logged in. i.e only admin users will see the 'Dashboard' link.

leechristophermurray committed:
Update db.js

reverted mongoose.connect() arguments.

leechristophermurray committed:
added comments to the '.isActive()' function

leechristophermurray commented:
@matjames007 this pull request may be reviewed again.

matjames007 commented:
@thinkking merge in `develop`.

Issues:

- iFrame is a bad way for presenting the documentation and presents additional issues when the page is scaled to a tablet display
- Instead of making one call to load the page, two calls are made to the server with twice the calls for all the page resources used by the application.  This means that an instance of the angular application is running with an instance of the docs.ejs page inside the iframe with almost all the same scripts, styling files and html elements.
- Please ensure that you cover all the changes you did in the commit messages.

This being said - i suggest you demonstrate the functionality to another core team member for feedback on whether this can be merge in as part of the MVP.

leechristophermurray committed:
localhost:27017/mongodb

matjames007 committed:
I opted for a more efficient way of doing the documentation.  Instead of an iframe I included it into the application.  The challenge of manipulating the DOM object within the HEAD tag was removed by utilizing an angular library that was capable of switching out the stylesheet based on the css specified within the $routeProvider.  It is important to note that the calls to the server are reduced and the UI interaction is preserved.

As a consequence however accomodations previously made to account for having the iFrame needs to be readjusted.

matjames007 committed:
Minor style changes to account for missing navigation active indicator.

matjames007 committed:
Creating a new Controller with the responsibility of initializing the Redoc elements.  This should give angular more control in the ordering of items it fires.  Testing needed.

-Matthew Budram

leechristophermurray committed:
commit before merge

leechristophermurray committed:
commit before merge

leechristophermurray committed:
setting the 'scroll-y-offset' in docs.html

leechristophermurray committed:
merging develop into the branch

leechristophermurray committed:
restyling the docs to accomodate the previous changes in structure

matjames007 commented:
Not sure what happened with those commits @thinkking .... but I think the .yaml file is broken and doesn't render well in redoc.

leechristophermurray committed:
modified the 'main.css for redoc to accomodate angular serving the doc spec. I also reverted the docs main description to the hack that was previously there

leechristophermurray commented:
@matjames007 and @nick22891 this pull request may be reviewed again. The docs are showing as they should due to a few lines of styling I had to do to the `main.css` we have for our docs.

matjames007 commented on models/db.js:
```diff
@@ -6,7 +6,7 @@
 var Schema = mongoose.Schema;
 
-mongoose.connect(process.env.MONGOLAB_URI);
+mongoose.connect(process.env.MONGOLAB_URI ||  "localhost:27017/mongodb");
```
@thinkking change again....

leechristophermurray committed:
Update db.js

Reverted the mongoose.connect() arguments.

leechristophermurray commented:
@matjames007 the mongoose.connect() arguments have been reverted.
